### PR TITLE
Pearl beefy strategy

### PR DESCRIPTION
**Pull request by maruf0011**
- [x] Strategy
- [x] Epoch Strategy
- [x] Test
- [x] Documentation
- [x] Simplification

### CODE REVIEW

File: src/strategies/BeefyPearlStrategy.sol [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        token0.approve(address(pearlRouter), type(uint256).max);
+        token1.approve(address(pearlRouter), type(uint256).max);
+
+        // approve lp token for pearlRouter and beefy vault
+        lpToken.approve(address(pearlRouter), type(uint256).max);
+        lpToken.approve(address(beefy), type(uint256).max);
+    }
+
+    // TODO: parameterize (address from, address to)
+    function _getInLPRatio() internal view returns (uint256, uint256) {
```
Comment (larryob) [inline src/strategies/BeefyPearlStrategy.sol:57]: I don't think this function is needed. See the comment on `_investIntoBeefy` below.

File: src/strategies/BeefyPearlStrategy.sol [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+            true,
+            lpTokenAmount,
+            minToken0Out,
+            minToken1Out,
+            address(this),
+            block.timestamp
+        );
+    }
+
+    function _investIntoBeefy(uint256 assets, uint256 slippage) internal {
```
[the remark this answers is not in the document]
Comment (maruf0011) [reply src/strategies/BeefyPearlStrategy.sol:134]: 1. Pool is not fully balanced, now its around 47/52 ratio. using half to swap will simplify but will have 500 bps of unused assets always. if the balance shifts utilization will go lower.

File: src/strategies/BeefyPearlStrategy.sol [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        if (assets <= asset.balanceOf(address(this))) {
+            return;
+        }
+
+        _divestLP(assets - asset.balanceOf(address(this)), slippage);
+
+        _swapToken(token1, token0, token1.balanceOf(address(this)), slippage);
+    }
+
+    function _divestLP(uint256 assets, uint256 slippage) internal {
```
[the remark this answers is not in the document]
Comment (maruf0011) [reply src/strategies/BeefyPearlStrategy.sol:201]: Yup this is what it does. Not sure about the concerns.

File: src/strategies/BeefyPearlStrategy.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (maruf0011) [reply src/strategies/BeefyPearlStrategy.sol:133]: as we are keeping this implementation, marking this as resolved.

**Pull request merged**